src/transformers/feature_extraction_sequence_utils.py

```diff
@@ -278,6 +308,47 @@ def _pad(
 
         return processed_features
 
+    def _truncate(
+        self,
+        processed_features: Union[Dict[str, List[float]], BatchFeature],
+        max_length: Optional[int] = None,
+        pad_to_multiple_of: Optional[int] = None,
+        truncation: Optional[bool] = None,
+    ):
+        """
+        Pad inputs (on left/right and up to predefined length or max length in the batch)
+
+        Args:
+            processed_features: Dictionary of input values (`List[float]`) / input vectors (`List[List[float]]`) or batch of inputs values (`List[List[int]]`) / input vectors (`List[List[List[int]]]`)
+            max_length: maximum length of the returned list and optionally padding length (see below)
+            pad_to_multiple_of: (optional) Integer if set will pad the sequence to a multiple of the provided value.
+                This is especially useful to enable the use of Tensor Core on NVIDIA hardware with compute capability
+                >= 7.5 (Volta), or on TPUs which benefit from having sequence lengths be a multiple of 128.
+            truncation: (optional) Activates truncation to cut input sequences longer than `max_length` to `max_length`.
+        """
+        if not truncation:
+            # no truncation
+            return processed_features
+        elif truncation and max_length is None:
+            raise ValueError(
+                "When setting ``truncation=True``, make sure that ``max_length`` is defined and ``padding='max_length'``"
```

Comment:
This implies that `padding='max_length'` is required when setting `truncation=True` but that's not the case, is it?

Reply:
`max_length` is actually required for truncation (see 3 use cases in the docstring above). We could allow `truncate=True` without any padding which would then truncate everything to the small sample in the batch, but this would be different from how we handle it in the tokenizers -> so I think throwing an error and forcing the user to define the `max_length` is the way to go here. @LysandreJik  What do you think?

Reply:
Ok yes this makes sense! LGTM